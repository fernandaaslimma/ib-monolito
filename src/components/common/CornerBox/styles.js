import styled, { css } from "styled-components";
import { rem } from "../../../styles/tools";

export const ContentWrapper = styled.div`
  ${({ width, height, cornerSize }) =>
    css`
      width: ${rem(width + cornerSize + cornerSize)};
      height: ${rem(height + cornerSize + cornerSize)};
    `};

  align-items: center;
  justify-content: center;
  flex-direction: column;

  ${({ cornerSize }) =>
    cornerSize &&
    css`
      padding: ${rem(cornerSize)};
    `};

  border: none;
  flex-wrap: wrap;
  position: relative;

  :after,
  :before {
    display: block;
    content: "";
    ${({ cornerSize, cornerThickness }) =>
      css`
        width: ${rem(cornerSize - cornerThickness)};
        height: ${rem(cornerSize - cornerThickness)};
      `};
    position: absolute;
  }

  :before {
    top: 0px;
    left: 0px;
    ${({ cornerColor, cornerThickness, cornerOpacity }) =>
      css`
        opacity: ${cornerOpacity};
        border-top: ${rem(cornerThickness)} solid ${cornerColor};
        border-left: ${rem(cornerThickness)} solid ${cornerColor};
      `};
  }
  :after {
    top: 0px;
    right: 0px;
    ${({ cornerColor, cornerThickness, cornerOpacity }) =>
      css`
        opacity: ${cornerOpacity};
        border-top: ${rem(cornerThickness)} solid ${cornerColor};
        border-right: ${rem(cornerThickness)} solid ${cornerColor};
      `};
  }
`;

export const Corner = styled.div`
  :after,
  :before {
    display: block;
    content: "";

    position: absolute;
    ${({ cornerSize, cornerThickness }) =>
      css`
        width: ${rem(cornerSize - cornerThickness)};
        height: ${rem(cornerSize - cornerThickness)};
      `};
  }

  :before {
    bottom: 0px;
    left: 0px;
    ${({ cornerColor, cornerThickness, cornerOpacity }) =>
      css`
        opacity: ${cornerOpacity};
        border-bottom: ${rem(cornerThickness)} solid ${cornerColor};
        border-left: ${rem(cornerThickness)} solid ${cornerColor};
      `};
  }
  :after {
    bottom: 0px;
    right: 0px;

    ${({ cornerColor, cornerThickness, cornerOpacity }) =>
      css`
        opacity: ${cornerOpacity};
        border-bottom: ${rem(cornerThickness)} solid ${cornerColor};
        border-right: ${rem(cornerThickness)} solid ${cornerColor};
      `};
  }
`;
