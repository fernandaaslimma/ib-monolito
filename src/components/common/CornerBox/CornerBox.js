import React, { Fragment } from "react";
import { string, node, number } from "prop-types";
import { ContentWrapper, Corner } from "./styles";

function CornerBox({
  width,
  height,
  cornerColor,
  cornerSize,
  cornerThickness,
  cornerOpacity,
  children
}) {
  return (
    <Fragment>
      <ContentWrapper
        width={width}
        height={height}
        cornerColor={cornerColor}
        cornerSize={cornerSize}
        cornerThickness={cornerThickness}
        cornerOpacity={cornerOpacity}
      >
        {children}
        <Corner
          cornerColor={cornerColor}
          cornerSize={cornerSize}
          cornerThickness={cornerThickness}
          cornerOpacity={cornerOpacity}
        ></Corner>
      </ContentWrapper>
    </Fragment>
  );
}

CornerBox.defaultProps = {
  width: 50,
  height: 50,
  cornerColor: "black",
  cornerSize: 5,
  cornerThickness: 2,
  cornerOpacity: 1,
  children: null
};

CornerBox.propTypes = {
  width: number,
  height: number,
  cornerColor: string,
  cornerSize: number,
  cornerThickness: number,
  cornerOpacity: number,
  children: node
};

export default CornerBox;
