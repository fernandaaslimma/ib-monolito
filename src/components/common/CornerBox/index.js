import CornerBox from "./CornerBox";

export default CornerBox;
