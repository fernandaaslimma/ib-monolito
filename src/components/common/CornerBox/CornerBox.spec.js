import React from "react";
import { shallow } from "enzyme";
import CornerBox from "./CornerBox";

const props = {
  current: 6,
  changeCurrentScreen: jest.fn()
};

describe("QrCode component", () => {
  it("should match snapshot with props", () => {
    expect(shallow(<CornerBox {...props} />)).toMatchSnapshot();
  });
});
