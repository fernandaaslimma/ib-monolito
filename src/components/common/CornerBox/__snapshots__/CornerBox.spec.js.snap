// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QrCode component should match snapshot with props 1`] = `
<Fragment>
  <styled.div
    cornerColor="black"
    cornerOpacity={1}
    cornerSize={5}
    cornerThickness={2}
    height={50}
    width={50}
  >
    <styled.div
      cornerColor="black"
      cornerOpacity={1}
      cornerSize={5}
      cornerThickness={2}
    />
  </styled.div>
</Fragment>
`;
